extra/datasets/wikipedia.py

```diff
@@ -0,0 +1,411 @@
+    input_ids.append(0)
+    input_mask.append(0)
+    segment_ids.append(0)
+  assert len(input_ids) == max_seq_length
+  assert len(input_mask) == max_seq_length
+  assert len(segment_ids) == max_seq_length
+
+  masked_lm_positions = instance["masked_lm_positions"]
+  masked_lm_ids = tokenizer.convert_tokens_to_ids(instance["masked_lm_labels"])
+  masked_lm_weights = [1.0] * len(masked_lm_ids)
+
+  while len(masked_lm_positions) < getenv("MAX_PREDICTIONS_PER_SEQ", 76):
+    masked_lm_positions.append(0)
+    masked_lm_ids.append(0)
+    masked_lm_weights.append(0.0)
+
+  next_sentence_label = 1 if instance["is_random_next"] else 0
+
+  return {
+    "input_ids": np.expand_dims(np.array(input_ids, dtype=np.int32), 0),
+    "input_mask": np.expand_dims(np.array(input_mask, dtype=np.int32), 0),
+    "segment_ids": np.expand_dims(np.array(segment_ids, dtype=np.int32), 0),
+    "masked_lm_positions": np.expand_dims(np.array(masked_lm_positions, dtype=np.int32), 0),
+    "masked_lm_ids": np.expand_dims(np.array(masked_lm_ids, dtype=np.int32), 0),
+    "masked_lm_weights": np.expand_dims(np.array(masked_lm_weights, dtype=np.float32), 0),
+    "next_sentence_labels": np.expand_dims(np.array([next_sentence_label], dtype=np.int32), 0),
+  }
+
+def process_part(part:int):
+  tokenizer = Tokenizer(getenv("BASEDIR", Path(__file__).parent / "wiki") / "vocab.txt")
+  os.makedirs(BASEDIR / "train" / str(part), exist_ok=True)
+  for i, feature_batch in enumerate(process_iterate(tokenizer, val=False, part=part)):
+    with open(BASEDIR / f"train/{str(part)}/{part}_{i}.pkl", "wb") as f:
+      pickle.dump(feature_batch, f)
+
+def process_iterate(tokenizer:Tokenizer, val:bool=False, part:int=0) -> list[dict]: # Convert raw text to masked NSP samples
+  rng = random.Random(getenv('RANDOM_SEED', 12345))
+
+  if val:
+    tqdm.write("Getting samples from dataset")
+    documents = get_documents(rng, tokenizer, "results4/eval.txt")
+    instances = get_instances(rng, tokenizer, documents)
+
+    tqdm.write(f"There are {len(instances)} samples in the dataset")
+    tqdm.write(f"Picking 10000 samples")
+
+    pick_ratio = len(instances) / 10000
+    picks = [instance_to_features(instances[int(inst*pick_ratio)], tokenizer) for inst in range(10000)]
+    for batch in range(10):
+      yield picks[batch*1000:(batch+1)*1000]
+  else:
+    documents = get_documents(rng, tokenizer, f"results4/part-{part:05d}-of-00500")
+    instances = get_instances(rng, tokenizer, documents)
+
+    while len(instances) > 0:
+      batch_size = min(1000, len(instances)) # We batch 1000 samples to one file
+      batch = instances[:batch_size]
+      del instances[:batch_size]
+      yield [instance_to_features(instance, tokenizer) for instance in batch]
+
+##################### Load files #####################
+
+def get_val_files(): return sorted(list((BASEDIR / "eval/").glob("*.pkl")))
+
+@diskcache
+def get_train_files(): return sorted(list((BASEDIR / "train/").glob("*/*.pkl")))
```

Comment:
`@diskcache` searches by name and there's already a `get_train_files` for imagenet.